MoltenVK/MoltenVK/OS/MTLRenderPipelineColorAttachmentDescriptor+MoltenVK.m

```diff
@@ -0,0 +1,99 @@
+/*
+ * MTLRenderPipelineDescriptor+MoltenVK.m
+ *
+ * Copyright (c) 2018-2023 Chip Davis
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include "MVKCommonEnvironment.h"
+
+#if MVK_USE_METAL_PRIVATE_API
+
+#include "MTLRenderPipelineDescriptor+MoltenVK.h"
+
+typedef NSUInteger MTLLogicOperation;
+// This is disgusting. We shouldn't have to do this. We can't even rely on this.
+typedef struct MTLRenderPipelineAttachmentDescriptorPrivate {
+	union {
+		struct {
+			BOOL blendingEnabled : 1;
+			MTLBlendOperation rgbBlendOperation : 3;
+			MTLBlendOperation alphaBlendOperation : 3;
+			MTLBlendFactor sourceRGBBlendFactor : 5;
+			MTLBlendFactor sourceAlphaBlendFactor : 5;
+			MTLBlendFactor destinationRGBBlendFactor : 5;
+			MTLBlendFactor destinationAlphaBlendFactor : 5;
+			MTLColorWriteMask writeMask : 4;
+			BOOL logicOpEnabled : 1;
+			MTLLogicOperation logicOp : 4;
+			MTLPixelFormat pixelFormat : 28;
+		};
+		struct {
+			uint64_t bits;
+		};
+	};
+} MTLRenderPipelineAttachmentDescriptorPrivate;
+
+@interface MTLRenderPipelineColorAttachmentDescriptor ()
+
+- (MTLRenderPipelineAttachmentDescriptorPrivate*)_descriptorPrivate;
+- (BOOL)isLogicOpEnabled;
+- (void)setLogicOpEnabled: (BOOL)enable;
+- (MTLLogicOperation)logicOp;
+- (void)setLogicOp: (MTLLogicOperation)op;
+
+@end
+
+@implementation MTLRenderPipelineColorAttachmentDescriptor (MoltenVK)
+
+- (BOOL)isLogicOpEnabledMVK {
+	if ([self respondsToSelector:@selector(isLogicOpEnabled)]) {
+		return [self isLogicOpEnabled];
+	}
+	if (![self respondsToSelector:@selector(_descriptorPrivate)]) {
+		return NO;
+	}
+	return [self _descriptorPrivate]->logicOpEnabled;
+}
+
```

Comment:
Is this saying there are cases where `logicOpEnabled` is actually a supported (private) Metal property, but other cases we have to reach into `_descriptorPrivate`? 

Is there rhyme or reason to it (eg. OS version, GPU type, etc)? Is it worth supporting the former, but not the latter (again, to reduce fragility)?

Reply:
> Is this saying there are cases where `logicOpEnabled` is actually a supported (private) Metal property, but other cases we have to reach into `_descriptorPrivate`?

Yes, pretty much.

> Is there rhyme or reason to it (eg. OS version, GPU type, etc)?

I believe it is based on OS version.

Reply:
> I believe it is based on OS version.

Yeah...probably. See my follow up note above. For now, until proven necessary by customer complaints, I'd feel better leaving `_descriptorPrivate` out, and just relying only on the `logicOpEnabled` property.